wallet/src/main/java/com/android/identity_credential/wallet/util/ActivityLogger.kt

```diff
@@ -0,0 +1,307 @@
+                                    requestType = requestType,
+//                                    deviceRequest = request,
+//                                    deviceResponse = response
+                                )
+                            }
+                        }
+                        entries.add(entry)
+                    }
+                }
+            }
+        }
+        return entries
+    }
+
+    fun deleteEntries(entries: List<ActivityEntry>) {
+        for (entry in entries) {
+            val key = STORAGE_KEY_PREFIX + entry.activityId
+            storageEngine.delete(key)
+            Logger.i(TAG, "Deleted entry with key: $key")
+        }
+    }
+
+    private fun CborMap.getString(key: String): String {
+        val cborObject = this[key] ?: return "" // Return empty string if key not found
+        return cborObject.asTstr.toString()
+    }
+
+    private fun CborMap.getLong(key: String): Long {
+        val cborObject = this[key] ?: throw IllegalArgumentException("Key $key not found")
+        return cborObject.asNumber.toLong()
+    }
+
+    private fun CborMap.getStringArray(key: String): Array<String>? {
+        val cborArray = this[key]?.asArray ?: return null
+        return cborArray.map { it.asTstr.toString() }.toTypedArray()
+    }
+
+    private fun preprocessJsonString(jsonString: String): String {
+        // Regular expression to match and remove 24(<< and >>) while keeping the remaining content
+        val pattern = Regex("""\d+\(\s*<<\s*([^>]*)\s*>>\s*\)""")
+        return jsonString.replace(pattern) { matchResult ->
+            matchResult.groupValues[1].trim()
+        }
+    }
+
+    fun getRequestedInfo(cborByteArray: ByteArray): Array<String> {
+        val jsonString = Cbor.toDiagnostics(cborByteArray, setOf(DiagnosticOption.EMBEDDED_CBOR))
+        val processedJsonString = preprocessJsonString(jsonString)
+
+        val identifiers = mutableListOf<String>()
+
+        try {
+            val jsonObject = JSONObject(processedJsonString)
+            val docRequests = jsonObject.getJSONArray("docRequests")
+
+            for (i in 0 until docRequests.length()) {
+                val docRequest = docRequests.getJSONObject(i)
+                val itemsRequest = docRequest.getJSONObject("itemsRequest")
+                val nameSpaces = itemsRequest.getJSONObject("nameSpaces")
+
+                if (nameSpaces.has("org.iso.18013.5.1")) {
+                    val elements = nameSpaces.getJSONObject("org.iso.18013.5.1")
+                    val elementKeys = elements.keys()
+
+                    while (elementKeys.hasNext()) {
+                        val elementKey = elementKeys.next()
+                        identifiers.add(elementKey)
+                    }
+                }
+            }
+        } catch (e: JSONException) {
+            Logger.e(TAG, "Error during requested info extraction", e)
+        }
+
+        return identifiers.toTypedArray()
+    }
```

Comment:
This seems like a complicated way of parsing a CBOR DeviceRequest...

Reply:
Fixed.